### PR TITLE
Update react-apollo, add @apollo/react-hooks, @apollo/react-common

### PR DESCRIPTION
**Pull request by aaronschachter**
### What does this PR do?

Branching off of #1694 -- this PR updates `react-apollo` to v3, and adds `@apollo/react-hooks` and  `@apollo/react-common` to get the `useApolloClient` hook working in the School Finder, per https://github.com/DoSomething/phoenix-next/pull/1694#pullrequestreview-315005673, and comments in the [Stack Overflow thread](https://stackoverflow.com/questions/57743313/no-apollo-client-instance-can-be-found/57743861#comment101926841_57743861) about the `Invariant Violation: No Apollo Client instance can be found` error returned when trying to use `useApolloClient` hook with current packages. 

### Any background context you want to provide?

Opening this as a separate PR from the School Finder + GraphQL work , as I'm not sure if there's anything this could/would break? cc @DFurnes 

### What are the relevant tickets/cards?

#1694

### Checklist

* [ ] Documentation added for new features/changed endpoints.
* [ ] Added appropriate feature/unit tests.
* [ ] Added screenshot to PR description of related front-end updates on **small** screens.
* [ ] Added screenshot to PR description of related front-end updates on **medium** screens.
* [ ] Added screenshot to PR description of related front-end updates on **large** screens.

### CODE REVIEW

File: resources/assets/components/App.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { Provider } from 'react-redux';
-import { ApolloProvider } from 'react-apollo';
+import { ApolloProvider } from '@apollo/react-common';
```
Comment (aaronschachter) [inline resources/assets/components/App.js]: One [commenter mentions](https://stackoverflow.com/questions/57743313/no-apollo-client-instance-can-be-found/57743861#comment101926808_57743861) that making this change "fixed the useApolloClient but broke some of my other pages". I'm going to test around various pages to see if I can find any breakage.

**Comment by aaronschachter** [comment]
Breakage found (thanks to Cypress test), the `Embed` components on the Beta Referral page is throwing an error for cannot find property `embed`. Pushing up a fix by checking whether the `Embed` is loaded.

Screen Shot 2019-11-11 at 11 15 31 AM

File: resources/assets/components/utilities/Embed/Embed.js [hunk 2]
```diff
@@ -55,6 +55,9 @@ const Embed = props => {
       <Query query={EMBED_QUERY} variables={{ url }}>
         {({ loading, error, data }) => {
           const isLoaded = !loading;
+          if (!isLoaded) {
+            return null;
+          }
           const { embed } = data;
```
Comment (DFurnes) [inline resources/assets/components/utilities/Embed/Embed.js]: Maybe we can re-write this as `get(data, embed, {})` so this shows our placeholder below.

File: package.json [hunk 3]
```diff
@@ -59,6 +59,9 @@
     "testURL": "http://localhost"
   },
   "dependencies": {
+    "@apollo/react-common": "^3.1.3",
+    "@apollo/react-hooks": "^3.1.3",
+    "@apollo/react-testing": "^3.1.3",
```
Comment (weerd) [inline package.json]: Should `@apollo/react-testing` be a dependency or should it be a dev-dependency? I suspect it's only used for testing right?

**Pull request merged**